framework/i18n/Formatter.php

```diff
@@ -578,16 +577,30 @@ public function asImage($value, $options = [])
      * Formats the value as a hyperlink.
      * @param mixed $value the value to be formatted.
      * @param array $options the tag options in terms of name-value pairs. See [[Html::a()]].
+     * @param bool|string $scheme the URI scheme to use in the formatted hyperlink (available since 2.0.43):
+     *
+     * - `false (default)`: adding non-secure protocol scheme if there is none added already
+     * - `true`: adding secure protocol scheme if there is none added already
+     * - string: adding the specified scheme (either `http`, `https` or empty string
+     *   for protocol-relative URL) if there is none added already
+     *
      * @return string the formatted result.
      */
-    public function asUrl($value, $options = [])
```

Comment:
This breaks BC since it changes method signature (it will generate fatal error if someone overrides this method: https://3v4l.org/rBKHk).

Maybe it would be better to pass scheme inside of `$options`?

Reply:
Also, why not use `Url::ensureScheme()`  here? Separate implementation creates code duplication and inconsistencies. Right now formatter dos not recognize protocol-relative URLs and may create something like `http:////example.com`.

Reply:
It's valid point but I think we can get away with that particular BC. I'll add upgrade note.
And about the ensureScheme - I've tried to add it but output was always somehow wrong, not sure what it was exactly. I've ended with this straight forward solution since this method is so plain anyway. If you have better solution please PR it.

Reply:
> It's valid point but I think we can get away with that particular BC.

It is strictly against our [BC policy](https://github.com/yiisoft/yii2/blob/master/docs/internals/bc.md). If nobody cares, then maybe it would be better to remove this document and do not spread false promises?

As for `Url::ensureScheme()`, maybe the "problem" was because it actually handles relative URL in correct way? That was (and still is) a real pain for this formatter, and a main reason why I almost always overwrite it to handle URLs like `/some/relative/path`.

Reply:
Well, it's not `2.x.y.Z`. But you are right, we tend to ignore BC breaks too often. @samdark should we rollback that additional method argument?

Reply:
@bizley up to you. I'm fine with either keeping it or reverting it.

Reply:
I'll rewrite it then and use `$options` instead and `Url::ensureScheme()` as @rob006 suggested.